samples/net/mdns_responder/src/main.c

```diff
@@ -10,8 +10,13 @@
 LOG_MODULE_REGISTER(net_mdns_responder_sample, LOG_LEVEL_DBG);
 
 #include <zephyr.h>
+#include <net/dns_sd.h>
 #include <net/net_core.h>
 
+static const uint16_t zephyr_tcp_port = sys_cpu_to_be16(4242);
+DNS_SD_REGISTER_TCP_SERVICE(zephyr_tcp, CONFIG_NET_HOSTNAME, "_zephyr",
+	"local", DNS_SD_EMPTY_TXT, &zephyr_tcp_port);
```

Comment:
After some pondering, I am not sure if this is a good idea after all. As the port number is just an integer in user API, it is actually quite confusing if user needs to supply an integer in network byte order. Can't we hide this inside the macro, so user would provide port number in host order, but the macro would store it in network byte order?

Reply:
The macro should really store a pointer in order to facilitate ephemeral ports.

Reply:
I agree, it is a bit of an annoyance.

The main issue to is that the port can be either `const` or not.

It would be fairly trivial to create `_CONST` variants of the macros that also declare and initialize a `static const uint16_t` for the port (hiding byte-swapping in the macro, as you suggested).

For using the non-`const` macro (i.e. the current one), I can't see any advantage in compile-time initialization, so I think zero-initializing it is fine. I'll update the PR and see if you agree with the way the code looks then.

Reply:
OK - so the `DNS_SD_REGISTER_SERVICE` macro is the "advanced" one, where a user can pass in a pointer to a `uint16_t` that will need byte-swapping, but the `DNS_SD_REGISTER_TCP_SERVICE` and `DNS_SD_REGISTER_UDP_SERVICE` macros now do all of that for you, declaring the `uint16_t` and byte-swapping it internally. I think that should cover both the "common" and "advanced" use cases.

Let me know if you think more changes are necessary, @jukkar. The only thing I have left is to add a trivial service that is bound to a port in `samples/net/mdns_responder`.

Reply:
> Let me know if you think more changes are necessary,

No, this looks perfectly fine how it is done now.